src/components/onboarding/steps/collector_intent/option.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from "react"
+import styled from "styled-components"
+
+import * as fonts from "../../../../assets/fonts"
+import Icon from "../../../icon"
+
+interface OptionProps {
+  href?: string
+  text?: string
+  onSelect: (selected: boolean) => void
+}
+
+interface LinkState {
+  selected: boolean
+}
+
+const IconContainer = styled.div`
+  width: 15px;
+  height: 15px;
+  background-color: black;
+  display: none;
+  border-radius: 50%;
+  float: right;
+  margin-right: 15px;
+  line-height: 11px;
+`
+
+const Link = styled.a`
+  display: block;
+  font-size: 14px;
+  color: black;
+  text-decoration: none;
+  text-transform: uppercase;
+  font-family: ${fonts.primary.fontFamily};
+  padding: 30px 0 30px 15px;
+  border-top: 1px solid #e5e5e5;
+  &:hover {
+    background-color: #f8f8f8;
+  }
+  &:hover .collector-intent-checked {
+    display: inline;
+  }
+  & .collector-intent-checked.is-selected {
+    display: inline;
+  }
+`
+
+class Option extends React.Component<OptionProps, LinkState> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      selected: false,
+    }
+    this.onSelect = this.onSelect.bind(this)
```

Comment:
No need to bind event methods in this way -- again, ES7 classProperties allow you to write:

```js
class Option extends React.Component<OptionProps, LinkState> {
  onSelect = (event) => {
    ...
  }
}
```
The fat arrow binds to the outer scope.

Reply:
This was not stage 3 until very recently, before then it was still causing issues (such as not working well with method decorators), so it would be great if we can first verify that all the compilers have ironed out bugs since it reached stage 3.

Reply:
This has been in TypeScript for a few years now, and you can verify it works by trying it in whatever compnoent :)

Reply:
The situation I’m referring to was just a month ago, though, and class properties only reached stage 3 a month or so ago as well.

Reply:
(See the ‘important’ section here https://github.com/artsy/emission/pull/693 /cc @mennenia)

Reply:
I am sad about this too

Reply:
Aw damn -- I see what you mean. We should keep an eye on this.

Reply:
;/